contracts/Aavegotchi/facets/AavegotchiLendingFacet.sol

```diff
@@ -0,0 +1,291 @@
+        rental.timeAgreed = block.timestamp;
+
+        uint256 tokenId = rental.erc721TokenId;
+        s.lentTokenIdIndexes[originalOwner][tokenId] = s.lentTokenIds[originalOwner].length;
+        s.lentTokenIds[originalOwner].push(tokenId);
+
+        if (rental.erc721TokenAddress == address(this)) {
+            LibAavegotchi.transfer(originalOwner, renter, tokenId);
+        } else {
+            // External contracts
+            IERC721(rental.erc721TokenAddress).safeTransferFrom(originalOwner, renter, tokenId);
+        }
+
+        // set original owner as pet operator
+        s.petOperators[renter][originalOwner] = true;
+
+        emit ERC721ExecutedRental(
+            _rentalId,
+            originalOwner,
+            renter,
+            rental.erc721TokenAddress,
+            tokenId,
+            rental.amountPerDay,
+            rental.period,
+            block.timestamp
+        );
+    }
+
+    ///@notice Allow a original owner to claim revenue from the rental
+    ///@dev Will throw if the NFT has not been lent or if the rental has been canceled already
+    ///@param _rentalId The identifier of the rental to claim
+    function claimAavegotchiRental(uint256 _rentalId) external {
+        AavegotchiRental storage rental = s.aavegotchiRentals[_rentalId];
+        require(rental.timeCreated != 0, "AavegotchiLending: rental not found");
+        require(rental.timeAgreed != 0, "AavegotchiLending: rental not agreed");
+        require(rental.completed == false, "AavegotchiLending: rental already completed");
+
+        address sender = LibMeta.msgSender();
+        address originalOwner = rental.originalOwner;
+        address renter = rental.renter;
+        address receiver = rental.receiver;
+        uint256 completionTime = rental.timeAgreed + rental.period * 1 days;
+        if (completionTime + 1 days > block.timestamp) {
+            require(originalOwner == sender, "AavegotchiLending: only owner can claim during agreement");
+        }
+
+        uint256 tokenId = rental.erc721TokenId;
+        address escrow = s.aavegotchis[tokenId].escrow;
+        address collateralType = s.aavegotchis[tokenId].collateralType;
+        for (uint256 i; i < s.revenueTokens.length; i++) {
+            address revenueToken = s.revenueTokens[i];
+            if (escrow == address(0)) continue;
+            if (collateralType == revenueToken) continue;
+
+            uint256 balance = IERC20(revenueToken).balanceOf(escrow);
+            if (balance == 0) continue;
+
+            if (IERC20(revenueToken).allowance(escrow, address(this)) < balance) {
+                CollateralEscrow(escrow).approveAavegotchiDiamond(revenueToken);
+            }
+            if ((completionTime + 1 days <= block.timestamp) && (sender != originalOwner)) {
+                uint256 senderAmount = (balance * 3) / 100;
+                LibERC20.transferFrom(revenueToken, escrow, sender, senderAmount);
+                balance = (balance * 97) / 100;
```

Comment:
I suggest using `balance -= senderAmount` because that won't leave any remainders from division.

Reply:
OK.